android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -895,18 +895,37 @@ Java_com_mapswithme_maps_Framework_nativeGetRouteFollowingInfo(JNIEnv * env, jcl
 }
 
 JNIEXPORT jintArray JNICALL
-Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height)
+Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height, jobject routeAltitudeLimits)
 {
   ::Framework * fr = frm();
   ASSERT(fr, ());
 
   vector<uint8_t> imageRGBAData;
-  if (!fr->GenerateRouteAltitudeChart(width, height, imageRGBAData))
+  int32_t minRouteAltitude = 0;
+  int32_t maxRouteAltitude = 0;
+  measurement_utils::Units units = measurement_utils::Units::Metric;
+  if (!fr->GenerateRouteAltitudeChart(width, height, imageRGBAData, minRouteAltitude, maxRouteAltitude, units))
   {
     LOG(LWARNING, ("Can't generate route altitude image."));
     return nullptr;
   }
 
+  // Passing route limits.
+  jclass const routeAltitudeLimitsClass = env->GetObjectClass(routeAltitudeLimits);
+  ASSERT(routeAltitudeLimitsClass, ());
+
+  static jfieldID const minRouteAltitudeField = env->GetFieldID(routeAltitudeLimitsClass, "minRouteAltitude", "I");
```

Comment:
On the one hand methods GetFieldID and GetMethodID are slow. ("In the worst case it has to process the strings, and lookup the fieldID in the corresponding class data."). Because of it in our project we always use static or global for GetFieldID and very ofter for GetMethodID. @yunikkk insists on it.

On the other hand according "to the specification, the field ID is guaranteed to be valid only as long as the class is loaded." (routeAltitudeLimitsClass) "So, if for some reason the class gets unloaded, and then loaded again, a former field ID is probably invalid. You can work around that by storing the jclass reference as a global reference somewhere, thus preventing the GC from unloading the class."

According to 
https://rkennke.wordpress.com/2007/07/24/efficient-jni-programming-ii-field-and-method-access/
we cannot just assign result of GetFieldID/GetMethodID to static value and reuse it laster.

@yunikkk Could I ask you please to comment it?